### PR TITLE
Add more details as to why a user would want to use nullIf

### PR DESCRIPTION
**Pull request by gem-neo4j**
There has been some confusion as to the use case of nullIf, this adds a bit more reasoning in the docs as to how a user might use it!

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
Thanks for the documentation updates.

The preview documentation has now been torn down - reopening this PR will republish it.

**Review by loveleif — APPROVED** [review]

File: modules/ROOT/pages/functions/scalar.adoc [hunk 1]
```diff
@@ -768,6 +769,34 @@ The first parameter, "abc", is returned, as the two parameters are not equivalen
...
+======
+
+.Query
+[source, cypher, indent=0]
+----
+MATCH (a)
+RETURN a.name AS name, coalesce(nullIf(a.eyes, "brown"), "hazel") AS eyeColor
+----
+
+The `nullIf()` function can be used in conjunction with the `coalesce()` function for transforming one data value into another value:
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/functions/scalar.adoc]: I think it makes sense for this to go before the query example (next suggestion is just copying this sentence)
```suggestion
```

File: modules/ROOT/pages/functions/scalar.adoc [hunk 2]
```diff
@@ -768,6 +769,34 @@ The first parameter, "abc", is returned, as the two parameters are not equivalen
 
 ======
 
+.+nullIf()+
+======
+
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/functions/scalar.adoc:774]:
```suggestion

The `nullIf()` function can be used in conjunction with the `coalesce()` function for transforming one data value into another value:

```

File: modules/ROOT/pages/functions/scalar.adoc [hunk 3]
```diff
@@ -692,6 +692,7 @@ The length of the path `p` is returned.
 == nullIf()
 
 The function `nullIf()` returns null if the two given parameters are equivalent, otherwise returns the value of the first parameter.
+This function is the opposite of the xref:functions/scalar.adoc#functions-coalesce[coalesce()] function, which returns a default value, if the given value is null.
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/functions/scalar.adoc]:
```suggestion
This function is the opposite of the xref:functions/scalar.adoc#functions-coalesce[coalesce()] function, which returns a default value if the given value is null.
```

**Review by JPryce-Aklundh — APPROVED** [review]
lgtm to me - just a few editorial points :)

**Pull request merged**